# HG changeset patch
# User valeriep
# Date 1361905960 28800
#      Tue Feb 26 11:12:40 2013 -0800
# Node ID 0af95754ffc5d63e2eeb472e3e87e8d8b7ed3048
# Parent  f94c09f6a10e9ba492b8ad2fc6fe0900bca4374f
8000897, PR2173, RH1155012: VM crash in CompileBroker
Summary: Fixed to use the corresponding digest length when generating output.
Reviewed-by: mullan

diff -r f94c09f6a10e -r 0af95754ffc5 src/share/classes/sun/security/provider/SHA2.java
--- openjdk/jdk/src/share/classes/sun/security/provider/SHA2.java	Thu Oct 16 17:30:05 2014 +0100
+++ openjdk/jdk/src/share/classes/sun/security/provider/SHA2.java	Tue Feb 26 11:12:40 2013 -0800
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2002, 2012, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2002, 2013, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -101,7 +101,7 @@
         i2bBig4((int)bitsProcessed, buffer, 60);
         implCompress(buffer, 0);
 
-        i2bBig(state, 0, out, ofs, 32);
+        i2bBig(state, 0, out, ofs, engineGetDigestLength());
     }
 
     /**
